### PR TITLE
Aerospike backend update

### PR DESCRIPTION
**Pull request by reugn**

### CODE REVIEW

**Comment by reugn** [comment]
@ncabatoff could you please assist with the changelog entry failure in tests.

**Comment by ncabatoff** [comment]
> @ncabatoff could you please assist with the changelog entry failure in tests.

Actually, in this case the check is correctly reporting a problem.  You're making changes to production code, which should be reflected in the changelog.  Please create a changelog entry in your PR, as per https://github.com/hashicorp/vault/blob/main/CONTRIBUTING.md#changelog-entries .

**Review by calvn — APPROVED** [review]

**Comment by calvn** [comment]
@reugn can you run `go mod tidy` to clean up the go.sum file?

File: go.mod [hunk 1]
```diff
@@ -53,7 +53,7 @@ require (
 	github.com/go-sql-driver/mysql v1.5.0
 	github.com/go-test/deep v1.0.7
 	github.com/gocql/gocql v0.0.0-20210401103645-80ab1e13e309
-	github.com/golang/protobuf v1.4.2
```
Comment (calvn) [inline go.mod:56]: @briankassouf or @ncabatoff do you see any potential issues with the protobuf bump here and below?

**Comment by calvn** [comment]
@reugn would it be possible for you to resolve the merge conflicts on the go.mod and go.sum files?

**Comment by reugn** [comment]
@calvn resolved.

**Comment by reugn** [comment]
@calvn what is the difference with #13008?

**Comment by calvn** [comment]
Hey @reugn we were trying to get this in for 1.9 but due to logistics and code freeze, this didn't make it. We wanted to get this updated all the way to v5.6.0 back then, but it looks like you've gone ahead and done that since, so #13008 can be closed. I am trying to see if we can get this in for 1.9.1.

**Comment by reugn** [comment]
Sure @calvn. Let me know if you need anything else before you merge this.

**Comment by calvn** [comment]
@reugn we should be good to have this in for 1.9.1, but just to double-check on this since there's been a lot of changes in go.mod/go.sum recently, can you merge main in and run `go mod tidy` once again?

File: .gitignore [hunk 2]
```diff
@@ -27,6 +27,7 @@ _testmain.go
 # Other dirs
 /bin/
 /pkg/
+/vendor
```
Comment (calvn) [inline .gitignore:30]: This shouldn't be necessary since we're not vendoring in the repo.

Comment (reugn) [reply .gitignore]: @calvn reverted.

**Pull request merged**